### DISCUSSION TITLE
How to update SSG website when headless CMS content has been updated?

### DISCUSSION THREAD

**Discussion by jacktan165** [vercel/next.js#12473] [Help] [answered]
Currently, I generate my static website, and I get its information from a headless CMS called Strapi.
If someone were to make a change in the headless CMS, what is the best way to rebuild the website without any developers touching it?

**Comment by brianlovin** [comment, marked as answer]
As best I can tell, the only current way is to use the `unstable_revalidate` key in `getStaticProps` and set some kind of short-duration revalidation value (once this is out of RFC it would just be the `revalidate` key). I've asked about something similar as well though, basically a webhook/API to programmatically trigger a `revalidate` rather than waiting on a time value.

More: https://github.com/zeit/next.js/discussions/11552

And in https://github.com/zeit/next.js/discussions/11552#discussioncomment-6571:
> we're also going to add a way to trigger the revalidation through eg a webhook in your cms.
